src/main/io/ledstrip.h

```diff
@@ -86,46 +123,42 @@ typedef struct specialColorIndexes_s {
     uint8_t color[LED_SPECIAL_COLOR_COUNT];
 } specialColorIndexes_t;
 
-typedef struct ledConfig_s {
-    uint8_t xy;     // see LED_X/Y_MASK defines
-    uint8_t color;  // see colors (config_master)
-    uint16_t flags; // see LED_FLAG_FUNCTION + LED_FLAG_DIRECTION
-} ledConfig_t;
+typedef uint32_t ledConfigPacked_t;
 
-#define LED_X_BIT_OFFSET 4
-#define LED_Y_BIT_OFFSET 0
-#define LED_XY_MASK      0x0F
+typedef struct ledCounts_s {
+    uint8_t ledCount;
+    uint8_t ledRingCount;
+    uint8_t ledLarsonScannerCount;
+    uint8_t ledRingSeqLen;
+} ledCounts_t;
 
-static inline int ledGetX(const ledConfig_t *lcfg) { return (lcfg->xy >> LED_X_BIT_OFFSET) & LED_XY_MASK; }
-static inline int ledGetY(const ledConfig_t *lcfg) { return (lcfg->xy >> LED_Y_BIT_OFFSET) & LED_XY_MASK; }
-static inline void ledSetXY(ledConfig_t *lcfg, int x, int y) {
-    lcfg->xy = ((x & LED_XY_MASK) << LED_X_BIT_OFFSET) | ((y & LED_XY_MASK) << LED_Y_BIT_OFFSET);
-}
-#define CALCULATE_LED_XY(x, y) ((((x) & LED_XY_MASK) << LED_X_BIT_OFFSET) | (((y) & LED_XY_MASK) << LED_Y_BIT_OFFSET))
+#define LED(x, y, col, dir, func, ol, params) LED_MOV_POS(CALCULATE_LED_XY(x, y)) | LED_MOV_COLOR(col) | LED_MOV_DIRECTION(dir) | LED_MOV_FUNCTION(func) | LED_MOV_OVERLAY(ol) | LED_MOV_PARAMS(params)
+
+static inline uint8_t ledGetXY(const ledConfigPacked_t* lcfg)         { return ((*lcfg >> LED_POS_OFFSET) & MASK(LED_POS_BITCNT)); }
+static inline uint8_t ledGetX(const ledConfigPacked_t* lcfg)          { return ((*lcfg >> (LED_POS_OFFSET + LED_X_BIT_OFFSET)) & LED_XY_MASK); }
+static inline uint8_t ledGetY(const ledConfigPacked_t* lcfg)          { return ((*lcfg >> (LED_POS_OFFSET + LED_Y_BIT_OFFSET)) & LED_XY_MASK); }
+static inline uint8_t ledGetFunction(const ledConfigPacked_t* lcfg)   { return ((*lcfg >> LED_FUNCTION_OFFSET) & MASK(LED_FUNCTION_BITCNT)); }
+static inline uint8_t ledGetOverlay(const ledConfigPacked_t* lcfg)    { return ((*lcfg >> LED_OVERLAY_OFFSET) & MASK(LED_OVERLAY_BITCNT)); }
+static inline uint8_t ledGetColor(const ledConfigPacked_t* lcfg)      { return ((*lcfg >> LED_COLOR_OFFSET) & MASK(LED_COLOR_BITCNT)); }
+static inline uint8_t ledGetDirection(const ledConfigPacked_t* lcfg)  { return ((*lcfg >> LED_DIRECTION_OFFSET) & MASK(LED_DIRECTION_BITCNT)); }
+static inline uint8_t ledGetParams(const ledConfigPacked_t* lcfg)     { return ((*lcfg >> LED_PARAMS_OFFSET) & MASK(LED_PARAMS_BITCNT)); }
 
-extern uint8_t ledCount;
-extern uint8_t ledRingCount;
+static inline bool maskGetBit(int mask, int id)                       { return ((mask >> id) & 1); }
```

Comment:
Actually, it's also used a couple time for directions. But I followed your idea and added:

```
static inline bool ledGetOverlayBit(const ledConfig_t *lcfg, int id) { return ((ledGetOverlay(lcfg) >> id) & 1); }
static inline bool ledGetDirectionBit(const ledConfig_t *lcfg, int id) { return ((ledGetDirection(lcfg) >> id) & 1); }
```